### PR TITLE
Feat/stochastic scheduling

### PR DESCRIPTION
**Pull request by giovannic**
An extention to allow models to do:

```r
api$schedule(
  event = infection, # individual::event object
  target = to_infect, # a vector of indices of individuals to trigger the event for
  delay = infection_times # when to trigger the events for each individual in `target`
)
```

Before you could only set one delay time.

There's an example in vignettes/Scheduling.Rmd.

## Changes:

 * Update SimAPI, process_api.cpp and Process.h to allow a vector of delays
 * Split Scheduler.schedule validation and execution
 * Overload Scheduler.schedule for a vector of delays
 * Add bitwise operations for IterableBitset
 * Test bitops on IterableBitset
 * Test scheduling a vector of delays

## Things to look out for:

 * Documentation suggestions
 * Anything really...

### CODE REVIEW

**Review by richfitz — COMMENTED** [review]
Two tiny comments here

**Comment by ceeb950** [comment]
All looks ok to me. Agree with Rich

**Comment by ceeb950** [comment]
Agree with Rich

File: inst/include/Scheduler.h [hunk 1]
```diff
@@ -125,27 +129,88 @@ inline individual_index_t Scheduler<TProcessAPI>::get_scheduled(
...
+    const std::string& event,
+    const std::vector<size_t>& target_vector,
+    const std::vector<double>& delay) {
+
     if (schedule_map.find(event) == schedule_map.end()) {
         Rcpp::stop("Unknown event");
     }
 
-    auto target_timestep = current_timestep + d;
+    //round the delays
```
[the remark this answers is not in the document]
Comment (giovannic) [reply inst/include/Scheduler.h]: I wanted to stay as true to the continuous values as possible when doing discrete scheduling.

Is it that the overhead of rounding would be unnecessary? I didn't want to assume how the delays would be sampled

[the remark this answers is not in the document]
Comment (giovannic) [reply inst/include/Scheduler.h]: Updated the code comment

[the remark this answers is not in the document]
Comment (richfitz) [reply inst/include/Scheduler.h]: > Is it that the overhead of rounding would be unnecessary? I didn't want to assume how the delays would be sampled

No, nothing about overhead, the explanation makes sense, thanks :)

**Pull request merged**